### PR TITLE
Enable the typos utility Github action

### PR DESCRIPTION
**Pull request by pdostal**
There is `_typos.toml` file where I correct all the false-positives and excluude all the third-party of autogenerated files.

Link to the project: [crate-ci/typos](https://github.com/crate-ci/typos)

### CODE REVIEW

File: lib/publiccloud/k8sbasetest.pm [hunk 1]
```diff
@@ -51,7 +51,7 @@ sub get_k8s_service_name {
         return "GKE";
     }
     elsif ($provider eq 'AZURE') {
-        return "AKS";
```
Comment (olblak) [inline lib/publiccloud/k8sbasetest.pm:54]: While I don't know the context here, it seems to me that AKS wasn't a typo

File: data/console/movies.redis [hunk 2]
```diff
@@ -92,7 +92,7 @@ HSET "movie:92" title "Honeytrap" genre "Crime" votes 325 rating 5.6 release_yea
 HSET "movie:93" title "X+Y" genre "Drama" votes 19609 rating 7.2 release_year 2014 
 HSET "movie:94" title "Creep" genre "Horror" votes 14696 rating 6.2 release_year 2014 
 HSET "movie:95" title "The Drop" genre "Crime" votes 109038 rating 7.1 release_year 2014 
-HSET "movie:96" title "Ich seh ich seh" genre "Drama" votes 22916 rating 6.7 release_year 2014 
+HSET "movie:96" title "Ich she ich she" genre "Drama" votes 22916 rating 6.7 release_year 2014
```
Comment (okurz) [inline data/console/movies.redis:95]: I suggest to also exclude this file

File: data/containers/Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # baseimage_var is a placeholder of your base image
-# use base variable when you call test_containered_app
+# use base variable when you call test_contained_app
```
Comment (okurz) [inline data/containers/Dockerfile:2]: wrong

File: data/hpc/installcheck_hpc_module.sh [hunk 4]
```diff
@@ -14,7 +14,7 @@ ftp://dist.suse.de/ibs/SUSE/Products/SLE-Module-Web-Scripting/12/${arch}/product
 base="ftp://openqa.suse.de/SLE-${version}-SERVER-POOL-${arch}-Media1-CURRENT/repodata/*-primary.xml.gz"
 sdk="ftp://openqa.suse.de/SLE-${version}-SDK-POOL-${arch}-Media1-CURRENT/repodata/*-primary.xml.gz"
 
-checkd=$(mktemp check-XXXXX)
+checked=$(mktemp check-XXXXX)
```
Comment (okurz) [inline data/hpc/installcheck_hpc_module.sh:17]: wrong

File: data/ldap/test.ldif [hunk 5]
```diff
@@ -5,7 +5,7 @@ member: cn=Puca Almeida,ou=Green Corporation,ou=People,dc=green,dc=com
 member: cn=Fredo Almeida,ou=LAB Technical,ou=People,dc=green,dc=com
 member: cn=Otavio Augustus,ou=Green Corporation,ou=People,dc=green,dc=com
 member: cn=Pedro de Lara,ou=Green Corporation,ou=People,dc=green,dc=com
-member: cn=Gugu Liberato,ou=LAB Technical,ou=People,dc=green,dc=com
+member: cn=Gugu Liberation,ou=LAB Technical,ou=People,dc=green,dc=com
```
Comment (okurz) [inline data/ldap/test.ldif:8]: wrong

File: lib/publiccloud/k8sbasetest.pm (hunk 1 above)
Comment (pdostal) [reply lib/publiccloud/k8sbasetest.pm:54]: Hello @olblak, true. I'm still working on this. There will be lot of such cases.

**Review by okurz — CHANGES_REQUESTED** [review]
good idea but too many wrong changes right now

File: tools/detect_unused_modules [hunk 6]
```diff
@@ -38,7 +38,7 @@ done
 for test in $MODULES ; do
     # Don't check non-existing modules it as it was either renamed or removed
     # also check only for tests in tests/ directory
-    git ls-files "*$test*" | grep -q ^tests/ || continue
+    git ls-files "*$test*" | grep -qE '^tests/.*\.pm' || continue
```
Comment (okurz) [inline tools/detect_unused_modules:41]: this looks like an unwanted change

**Review by okurz — CHANGES_REQUESTED** [review]
After skimming over the content I found one issue https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16125#discussion_r1059915718 which seems to be a conflict resolution gone wrong.

In general I like the approach to fix and prevent spelling mistakes. However in general I prefer if we use tooling which is less reliant on github actions for two reasons:
1. Contributors should be able to detect and fix spelling mistakes *before* they open a pull request
2. We should prevent a "vendor lock-in" which is making any switch of project repository or CI harder

Of course one can still call a spell checker as part of github actions but it would be preferred to just call an application directly which also contributors can call instead of just a github action

**Review by baierjan — CHANGES_REQUESTED** [review]
Indeed, it is a good idea. After all, it was also able to find and fix one real bug.

On the other hand, it still makes a lot of false-positives some of them pretty dangerous with serious impact. I tried to point some of the controversial suggestions.

**Comment by pdostal** [comment]
crate-ci/typos#642, crate-ci/typos#643, crate-ci/typos#316

**Review by d3flex — CHANGES_REQUESTED** [review]

**Pull request closed without merging**